l2geth/internal/ethapi/api.go

```diff
@@ -798,14 +798,19 @@ func DoCall(ctx context.Context, b Backend, args CallArgs, blockNrOrHash rpc.Blo
 	}
 	// Set sender address or use a default if none specified
 	var addr common.Address
-	if args.From == nil {
-		if wallets := b.AccountManager().Wallets(); len(wallets) > 0 {
-			if accounts := wallets[0].Accounts(); len(accounts) > 0 {
-				addr = accounts[0].Address
+	if rcfg.UsingOVM {
+		addr = common.HexToAddress("0x0000000000000000000000000000000000000000")
+	} else {
+		// Set sender address or use a default if none specified
+		if args.From == nil {
```

Comment:
I think the `if` statement that you added here should be inside of the `if args.From == nil {` block instead of outside of it. Otherwise looking good!

Reply:
Totally, thanks ! I'm still working my way through the mono repo, trying to debug and run (successfully) everything :)